helper/builtinplugins/builtin.go

```diff
@@ -0,0 +1,38 @@
+package builtinplugins
+
+import (
+	"github.com/hashicorp/vault/plugins/database/mysql"
+	"github.com/hashicorp/vault/plugins/database/postgresql"
+)
+
+type BuiltinFactory func() (interface{}, error)
+
+var BuiltinPlugins *builtinPlugins = &builtinPlugins{
+	plugins: map[string]BuiltinFactory{
+		"mysql-database-plugin":      mysql.New,
+		"postgresql-database-plugin": postgresql.New,
+	},
+}
+
+// The list of builtin plugins should not be changed by any other package, so we
+// store them in an unexported variable in this unexported struct.
+type builtinPlugins struct {
```

Comment:
The variable itself is exported, so a package could change it to point to `nil`. Minor, but still there. A way to get around this would be to just provide a package-level accessor function for an unexported variable (or, have all functions operate at the package level rather than be functions of the object).